src/functions/game/onWinRound.ts

```diff
@@ -0,0 +1,57 @@
+import { Handler } from 'aws-lambda';
+import { LambdaEventBody, WebSocketAPIGatewayEvent } from '../../types/event';
+import { WebSocketClient } from '../../websocket/WebSocketClient';
+import { Logger } from '../../utils/Logger';
+import { LambdaEventBodyPayloadOptions } from '../../types/payload';
+import { getCurrentDealer, startNewGameRound } from '../../dynamodb/gameStateDBService';
+import { getUsersInGame } from '../../dynamodb/gameDBService';
+import { User } from '../../models/User';
+import {
+  broadcastWinningTiles,
+  broadcastUpdateGameState,
+  broadcastGameReset,
+} from '../../websocket/broadcast/gameBroadcast';
+import { getConnectionIdsFromUsers } from '../../utils/broadcastHelper';
+import { GameState } from '../../models/GameState';
+
+/**
+ * Handler for Winning Round
+ * @param {WebSocketAPIGatewayEvent} event Websocket API gateway event
+ */
+export const handler: Handler = async (event: WebSocketAPIGatewayEvent): Promise<void> => {
+  Logger.createLogTitle('onWinRound.ts');
+
+  const { connectionId } = event.requestContext;
+  const body: LambdaEventBody = JSON.parse(event.body);
+  const { payload }: { payload: LambdaEventBodyPayloadOptions } = body;
+  const gameId = payload.gameId as string;
+  const winningTiles = (payload.tiles as unknown) as string[];
+  const ws = new WebSocketClient(event.requestContext);
+
+  try {
+    // TODO: Deal with errors if gameid is invalid
+    const gameState = await Promise.all([getCurrentDealer(gameId), getUsersInGame(gameId)]);
+    const dealer = gameState[0] as number;
+    const users = gameState[1] as User[];
+    const connectionIds = getConnectionIdsFromUsers(users);
+
+    // send winning tiles to all connections
+    await broadcastWinningTiles(ws, connectionIds, connectionId, winningTiles);
+
+    const updatedGameState = (await startNewGameRound(
+      gameId,
+      connectionIds,
+      users[dealer].connectionId !== connectionId, // change dealer if winner is not currently a dealer
+    )) as GameState;
```

Comment:
I would avoid type assertion here because what if the response (`updatedGameState `) is empty; it would still be broadcasted which waste some resources (but for testing type assertion should be fine 😂 ). 

I would do this:
```ts
    const updatedGameState = await startNewGameRound(
      gameId,
      connectionIds,
      users[dealer].connectionId !== connectionId, // change dealer if winner is not currently a dealer
    );

    if (updatedGameState) {
      // send current dealer and wind to all connections
      await broadcastUpdateGameState(ws, connectionIds, updatedGameState.dealer, updatedGameState.currentWind);

      // start new round and send new hands to user
      setTimeout(async () => {
        await broadcastGameReset(ws, connectionIds, updatedGameState);
      }, 5000);

      return response(200, 'New round started successfully');
    }
    
   console.error('updatedGameState is empty, please double check')
    return response(400, 'Updated game state is empty, failed to run onWinRound');
```
let me know if this makes sense to you 😃

Reply:
Makes sense, called type guard statement in typescript. I did it a little differently by checking if it is undefined and throwing an error if it is to the catch block below instead of wrapping the whole thing in an if statement.

Reply:
nitpicking. IDE gives me a warning, but it should be fine
![image](https://user-images.githubusercontent.com/21091475/90662100-cb7a0080-e1fc-11ea-9a84-627b0f37b3c5.png)